### DISCUSSION TITLE
May 2020: OpenSMTPD 6.7.1p1 release, table-procexec and many PoCs · poolp.org

### DISCUSSION THREAD

**Discussion by poolpOrg** [poolpOrg/poolp.org#120] [Articles]

**Comment by K14D** [comment]
Thanks for the great work and the informative write-up. table-procexec looks really promising. Packages like the Go and Python examples will make using this functionality a breeze. I am really looking forward to see this in a production ready state.

**Comment by risner** [comment]
New install, finally got working basic config but it looks different than all the guides.
smtpd.conf:
...
table credentials passwd:/usr/local/etc/mail/passwd
listen on lo0
listen on em0 port smtp filter { check_dyndns, check_rdns, check_fcrdns, senderscore, rspamd }
listen on em0 port smtps tls-require pki "mail" filter { check_dyndns, check_rdns, check_fcrdns, senderscore, rspamd }
listen on em0 port submission tls-require pki "mail" received-auth auth <credentials>
action "local_mail" mbox alias <aliases>
action "virtual_mail" lmtp "/var/run/dovecot/lmtp" rcpt-to virtual <virtuals>
action "outbound" relay
match from any mail-from <blackhole> for any reject
match from local for local action "local_mail"
match from any for domain <domains> action "virtual_mail"
match from local for any action "outbound"
match from any auth <credentials> for any action "outbound"

Dovecot:
passdb {
  args = /usr/local/etc/mail/passwd
  driver = passwd-file
}
disable_plaintext_auth=yes
ssl = yes
ssl_cert=/usr/local/etc/mail/full.pem
ssl_key=/usr/local/etc/mail/key.pem

**Comment by risner** [comment]
Now chasing sievec issues:
report-ham: line 1: error: require command: unknown Sieve capability `vnd.dovecot.pipe'.
report-ham: line 1: error: require command: unknown Sieve capability `imapsieve'.
report-ham: line 15: error: unknown command 'pipe' (only reported once at first occurrence).
report-ham: error: validation failed.
sievec(risner): Fatal: failed to compile sieve script 'report-ham.sieve'
